app/javascript/app/FeedbackForm.js

```diff
@@ -41,13 +41,38 @@ function FeedbackForm() {
     return this;
   }
 
+  function feedbackFormSend() {
+    const agent = navigator.userAgent;
+
+    const csrfToken = document.querySelector('meta[name="csrf-token"]').getAttribute('content');
+
+    const transmission = {
+      message: _commentInput.value,
+      from: _emailInput.value,
+      agent,
+    };
+
+    const xhr = new XMLHttpRequest();
+    xhr.open('POST', '/feedback');
+    xhr.setRequestHeader('Content-Type', 'application/json');
+    xhr.setRequestHeader('X-CSRF-Token', csrfToken);
+    xhr.onload = function() {
+      if (xhr.status === 200) {
+        _onSuccess();
+      } else if (xhr.status !== 200) {
+        _onError();
+      }
+    };
+    xhr.send(JSON.stringify(transmission));
+  }
+
   function _sendBtnClicked(evt) {
     // Stop the form from submitting.
     evt.preventDefault();
     var emailCheck = new RegExp('.+@.+\..+','i');
     var match = emailCheck.exec(_emailInput.value);
     if (match || _emailInput.value === '')
-      _feedbackFormSend();
+      feedbackFormSend();
```

Comment:
Underscore prefixing is a convention to show which methods are meant to be publicly exposed. Since `init` is the only method attached to the instance when it is returned, that's why it's the only one that doesn't have a leading underscore. It probably should be left as `_feedbackFormSend`

Reply:
The ES6 linter complained about the underscore. Does this convention still exist in ES6 or is there another way to define public methods?

Reply:
I found this in the ESLint docs: https://eslint.org/docs/rules/no-underscore-dangle. We are using the Airbnb version of ESLint, which does not allow the underscore prefix: https://github.com/airbnb/javascript/blob/a100a3957ffb11c841dbd74a649cbb9b5a17a2aa/packages/eslint-config-airbnb-base/rules/style.js#L358-L364

Reply:
Ah, so it must complain on all the other functions in the same file that have a leading underscore? Like the rule says, it's purely convention. Here's airbnb's reasoning, which sounds valid https://github.com/airbnb/javascript/issues/1408. Personally, I use them for when scanning the file to see what the public api (for tests and such) should be, but if you want to remove them all that's fine too. Thanks again!